### PR TITLE
ROSCI infra for live camera testing

### PR DESCRIPTION
**Pull request by kadiredd**
This rosci.py module is to enable test infra for ROSCI targeting live camera testing.
 - Detects devices connected and run ROS live tests on them by means of pytest-3
 - report test results in xml format by Junit xml
 -  Integrated test results xml with jenkins by JUnit utility and populate the results, analyze reports, and stats

### CODE REVIEW

File: realsense2_camera/test/live_camera/rosci.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+handle = None
+test_ran = False
+device_set = list()
+
+def usage():
+    ourname = os.path.basename( sys.argv[0] )
+    print( 'Syntax: ' + ourname + ' [options] ' )
+    print( 'Options:' )
+    print( '        -h, --help      Usage help' )
+    print( '            --debug     Turn on debugging information (does not include LibRS debug logs; see --rslog)' )
```
Comment (Arun-Prasad-V) [inline realsense2_camera/test/live_camera/rosci.py]: @kadiredd, does the `--debug` param being utilized anywhere? `--rslog`?

File: realsense2_camera/test/live_camera/rosci.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    print( '        -r, --regex     Run all tests whose name matches the following regular expression' )
+    print( '        --device <>     Run only on the specified device; list of devices separated by ',', No white spaces' )
```
Comment (Arun-Prasad-V) [inline realsense2_camera/test/live_camera/rosci.py]: @kadiredd, please also mention what happens if `--device` param is not passed.

File: realsense2_camera/test/live_camera/rosci.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+test_ran = False
+device_set = list()
+
+def usage():
+    ourname = os.path.basename( sys.argv[0] )
+    print( 'Syntax: ' + ourname + ' [options] ' )
+    print( 'Options:' )
+    print( '        -h, --help      Usage help' )
+    print( '            --debug     Turn on debugging information (does not include LibRS debug logs; see --rslog)' )
+    print( '        -r, --regex     Run all tests whose name matches the following regular expression' )
```
Comment (Arun-Prasad-V) [inline realsense2_camera/test/live_camera/rosci.py:48]: @kadiredd, please provide examples as well.

File: realsense2_camera/test/live_camera/rosci.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            log.i('Connected devices:', connected_devices)        
+        
+        testname = regex if regex else None
+        
+        if device_set:
+            for device in device_set:
+                if device.upper() in connected_devices:
+                    log.i('Running tests on device:', device)
+                    #for testname in test_list:
+                    cmd = command(device.lower(), testname)
```
Comment (Arun-Prasad-V) [inline realsense2_camera/test/live_camera/rosci.py:148]: @kadiredd, will it work if we send `regex` directly to `testname`?

File: realsense2_camera/test/live_camera/rosci.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    print( '        -h, --help      Usage help' )
+    print( '            --debug     Turn on debugging information (does not include LibRS debug logs; see --rslog)' )
+    print( '        -r, --regex     Run all tests whose name matches the following regular expression' )
+    print( '        --device <>     Run only on the specified device; list of devices separated by ',', No white spaces' )
+    print( '                        ex: --device=d455,d435i,d585 (or) --device d455,d435i,d585 ')
+        
+    sys.exit( 2 )
+
+def command(dev_name, test=None):
+       cmd =  ['pytest-3']
```
Comment (Arun-Prasad-V) [inline realsense2_camera/test/live_camera/rosci.py]: @kadiredd, please maintain the indentation throughout the whole file. In some places its 4 spaces and in some places more than 4.

File: realsense2_camera/test/live_camera/rosci.py (hunk 3 above)
Comment (kadiredd) [reply realsense2_camera/test/live_camera/rosci.py:48]: fixed

File: realsense2_camera/test/live_camera/rosci.py (hunk 4 above)
Comment (kadiredd) [reply realsense2_camera/test/live_camera/rosci.py:148]: Yes, this `regex` is either full test name or keyword and the same is appended in pytest command

File: realsense2_camera/test/live_camera/rosci.py (hunk 1 above)
Comment (kadiredd) [reply realsense2_camera/test/live_camera/rosci.py]: fixed

File: realsense2_camera/test/live_camera/rosci.py (hunk 2 above)
Comment (kadiredd) [reply realsense2_camera/test/live_camera/rosci.py]: Done

File: realsense2_camera/test/live_camera/rosci.py (hunk 3 above)
Comment (kadiredd) [reply realsense2_camera/test/live_camera/rosci.py:48]: done

File: realsense2_camera/test/live_camera/rosci.py (hunk 5 above)
Comment (kadiredd) [reply realsense2_camera/test/live_camera/rosci.py]: done

File: realsense2_camera/test/live_camera/rosci.py [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+test_ran = False
+device_set = list()
+
+def usage():
+    ourname = os.path.basename( sys.argv[0] )
+    print( 'Syntax: ' + ourname + ' [options] ' )
+    print( 'Options:' )
+    print( '        -h, --help      Usage help' )
+    print( '        -r, --regex     Run all tests whose name matches the following regular expression' )
+    print( '                        ex: --regex test_camera_imu; -r d415_basic')
```
Comment (SamerKhshiboun) [inline realsense2_camera/test/live_camera/rosci.py]: I would prefer "example:" or "e.g.,:", not "ex:"

File: realsense2_camera/test/live_camera/rosci.py [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+    print( '        --device <>     Run only on the specified device(s); list of devices separated by ',', No white spaces' )
+    print( '                        ex: --device=d455,d435i,d585 (or) --device d455,d435i,d585 ')
```
Comment (SamerKhshiboun) [inline realsense2_camera/test/live_camera/rosci.py]: same here (ex)

**Review by SamerKhshiboun — COMMENTED** [review]
Looks good to me, only small comment..

**Review by Nir-Az — APPROVED** [review]
Looks like a good point to start with

File: realsense2_camera/test/live_camera/rosci.py (hunk 6 above)
Comment (kadiredd) [reply realsense2_camera/test/live_camera/rosci.py]: fixed

**Review by Arun-Prasad-V — APPROVED** [review]

**Review by SamerKhshiboun — APPROVED** [review]

**Comment by kadiredd** [comment]
will re-open in next minute. closing and re-opening is for re-triggering the CI as I do not have permission to do that.

**Pull request merged**